### ISSUE TITLE
Yi-34B + Simpo + Full + Novel writing Task, fine-tuning results are unsatisfactory

### ISSUE THREAD

**Issue by onlyfish79** [princeton-nlp/SimPO#33]
* **Use LLama_Factory Project**： https://github.com/hiyouga/LLaMA-Factory
* **System Info**
```
llamafactory version: 0.8.3.dev0
Platform: Linux-6.5.0-41-generic-x86_64-with-glibc2.35
Python version: 3.10.13
PyTorch version: 2.2.1+cu121 (GPU)
Transformers version: 4.41.2
Datasets version: 2.18.0
Accelerate version: 0.31.0
PEFT version: 0.11.1
TRL version: 0.9.4
GPU type: NVIDIA A100-SXM4-80GB
DeepSpeed version: 0.13.4
Bitsandbytes version: 0.43.1
vLLM version: 0.4.1
```
* **执行脚本**
```
torchrun --nproc_per_node=8 --master_port=20001 src/train.py \
--stage dpo \
--pref_loss simpo \
--pref_beta 2.0 \
--simpo_gamma 1.0 \
--do_train \
--model_name_or_path /data/basemodel/Yi-34B \
--dataset novel \
--val_size 0.05 \
--dataset_dir /data/train_data \
--overwrite_cache \
--template yi \
--finetuning_type full \
--output_dir /data/dpo_model/yi34b_simpo_full \
--overwrite_output_dir \
--preprocessing_num_workers 128 \
--per_device_train_batch_size 4 \
--per_device_eval_batch_size 8 \
--gradient_accumulation_steps 4 \
--eval_strategy steps \
--eval_steps 10 \
--save_strategy steps \
--save_total_limit 1 \
--lr_scheduler_type cosine \
--logging_steps 1 \
--save_steps 230 \
--learning_rate 1e-5 \
--num_train_epochs 1.0 \
--max_grad_norm 0.5 \
--weight_decay 0 \
--warmup_ratio 0.03 \
--cutoff_len 2048 \
--plot_loss \
--bf16 \
--flash_attn fa2 \
--gradient_checkpointing \
--ddp_timeout 180000000 \
--deepspeed deepspeed/ds_zero3_offload_optimizer_param.json
```

* **ds_zero3_offload_optimizer_param.json**
```
{
  "bf16": {
    "enabled": "auto"
  },
  "optimizer": {
    "type": "AdamW",
    "params": {
      "lr": "auto",
      "betas": "auto",
      "eps": "auto",
      "weight_decay": "auto"
    }
  },
  "scheduler": {
    "type": "WarmupDecayLR",
    "params": {
      "total_num_steps": "auto",
      "warmup_min_lr": "auto",
      "warmup_max_lr": "auto",
      "warmup_num_steps": "auto"
    }
  },
  "zero_optimization": {
    "stage": 3,
    "offload_optimizer": {
      "device": "cpu",
      "pin_memory": true
    },
    "offload_param": {
      "device": "cpu",
      "pin_memory": true
    },
    "overlap_comm": true,
    "contiguous_gradients": true,
    "sub_group_size": 1e9,
    "reduce_bucket_size": "auto",
    "stage3_prefetch_bucket_size": "auto",
    "stage3_param_persistence_threshold": "auto",
    "stage3_max_live_parameters": 1e9,
    "stage3_max_reuse_distance": 1e9,
    "stage3_gather_16bit_weights_on_model_save": true
  },
  "gradient_accumulation_steps": "auto",
  "gradient_clipping": "auto",
  "steps_per_print": 50,
  "train_batch_size": "auto",
  "train_micro_batch_size_per_gpu": "auto",
  "wall_clock_breakdown": false
}
```

* **Data Set Fields**:  instruction，chosen，rejected

* **Task:** Novel writing
  - **Input**: Provide a novel excerpt
  - **Output**: 2-3 branch options

* **Problem Description**
We have a consistent environment, dataset, and base model. Using the **DPO** method, the fine-tuning results are **positive**. However, when using **SIMPO (both full parameter and LoRA fine-tuning),** the fine-tuned model generates poor results.
![image](https://github.com/princeton-nlp/SimPO/assets/18341845/6f4c72e8-c000-4a93-8d6e-c6e2487856c0)

**Expected behavior**
![image](https://github.com/princeton-nlp/SimPO/assets/18341845/110723d5-8239-4669-af44-2abb746f0aa2)

* **Question**
Is SIMPO unsuitable for novel writing tasks?

**Comment by xiamengzhou** [comment]
Hi @onlyfish79 Thanks for trying out SimPO in your task! I noticed that you used a relatively large learning rate for tuning, i.e., `1e-5`. In our experiments, we used much smaller learning rates for tuning and noticed that using a large learning rate will cripple results significantly. Please refer to table 8 in our [preprint](https://arxiv.org/pdf/2405.14734). We used a learning rate `3e-7`, `5e-7`, `6e-7`, `1e-6` for four settings respectively. I suggest that you grid search the grid learning properly.

**Comment by onlyfish79** [comment]
Thanks! I really appreciate you taking the time to read through my work in detail and provide such thoughtful advice.

I will definitely take your suggestions into consideration and experiment with smaller learning rates. Once I have some results, I will be sure to share them with you.

**Comment by onlyfish79** [comment]
> Hi @onlyfish79 Thanks for trying out SimPO in your task! I noticed that you used a relatively large learning rate for tuning, i.e., `1e-5`. In our experiments, we used much smaller learning rates for tuning and noticed that using a large learning rate will cripple results significantly. Please refer to table 8 in our [preprint](https://arxiv.org/pdf/2405.14734). We used a learning rate `3e-7`, `5e-7`, `6e-7`, `1e-6` for four settings respectively. I suggest that you grid search the learning rate properly.

I have refered to table 8 in our [preprint](https://arxiv.org/pdf/2405.14734)
![image](https://github.com/user-attachments/assets/f437d7c8-4acd-4c7f-88d7-514e64a0bb4d)

**Conclusion**: **I use β=2.0, γ=1.0, lr=6e-7**, The fine-tuned model generates results with correct content and format.

Thanks for your help!
